packages/lib/src/createEventEmitter/createEventEmitter.types.ts

```diff
@@ -0,0 +1,44 @@
+import type {Errors, SubmitParams} from '../FormControllerClass/FormControllerClass.types';
+
+export type FormagusEvent =
+  | {
+      type: 'submit:begin';
+    }
+  | ({
+      type: 'submit:end';
+    } & Omit<SubmitParams, 'event'>)
+  | {
+      type: 'validate:begin';
+    }
+  | {
+      type: 'validate:end';
+      errors: Errors;
+    }
+  | {
+      type: 'validateField:begin';
+      name: string;
+    }
+  | {
+      type: 'validateField:end';
+      name: string;
+      errors: Errors;
+    };
+
+export type InnerEventEmitter = {
```

Comment:
Had to define it instead of ReturnType<typeof createEventEmitter> as it creates circular dependency